core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -635,17 +636,21 @@ public static boolean hasAlleleFrequencyData (int geneticProfileId, int sampleId
             pstmt.setString(6, event.getProteinChange());
             pstmt.setString(7, event.getMutationType());
             rs = pstmt.executeQuery();
-            if (rs.next()) {
-                return extractMutationEvent(rs);
-            }
-            else {
-                return null;
+            while(rs.next()) {
+                mutationEventList.add(extractMutationEvent(rs));
```

Comment:
this is ok .. but there is some overhead in calling extractMutationEvent() on each returned record.
If the resultSet is pointing at a record where the TumorSeqAllele value differs from the string in the argument (event), we will not be returning it below ... so why do the construction of a model object and the lookup of a gene from the hashmap in DaoGeneOptimized? Also, here we are making a list which must be searched below. That can be avoided by putting the logic here:
String queryTumorSeqAllele = event.getTumorSeqAllele();
if (queryTumorSeqAllele == null) {
    queryTumorSeqAllele = "";
}
while(rs.next()) {
    String retrievedTumorSeqAllele = rs.getString("TUMOR_SEQ_ALLELE");
    if (retrievedTumorSeqAllele == null) {
        retrievedTumorSeqAllele = "";
    }
    if (queryTumorSeqAllele.equals(retrievedTumorSeqAllele)) {
        return extractMutationEvent(rs);
    }
}
return null;

Reply:
@sheridancbio thanks, I updated to return early if a match is found now instead of searching through a list at the end.